vortex-array/src/compute/list.rs

```diff
@@ -0,0 +1,127 @@
+//! List-related compute operations.
+
+use arrow_buffer::BooleanBuffer;
+use num_traits::AsPrimitive;
+use vortex_buffer::Buffer;
+use vortex_dtype::{NativePType, match_each_integer_ptype};
+use vortex_error::{VortexResult, vortex_bail};
+use vortex_mask::Mask;
+use vortex_scalar::Scalar;
+
+use crate::arrays::{ConstantArray, ListArray};
+use crate::compute::{Operator, compare};
+use crate::variants::PrimitiveArrayTrait;
+use crate::{Array, ArrayRef, IntoArray, ToCanonical};
+
+/// Get a `Mask` representing the positions in `array` that contains the scalar `value`.
+///
+/// ## Example
+///
+/// ```rust
+/// use vortex_array::{Array, IntoArray};
+/// use vortex_array::arrays::{ListArray, VarBinArray};
+/// use vortex_array::compute::list_contains;
+/// use vortex_array::validity::Validity;
+/// use vortex_buffer::buffer;
+/// use vortex_dtype::DType;
+/// let elements = VarBinArray::from_vec(
+///         vec!["a", "a", "b", "a", "c"], DType::Utf8(false.into())).into_array();
+/// let offsets = buffer![0u32, 1, 3, 5].into_array();
+/// let list_array = ListArray::try_new(elements, offsets, Validity::NonNullable).unwrap();
+///
+/// let matches = list_contains(&list_array, "b".into()).unwrap();
+/// let to_vec: Vec<bool> = matches.to_boolean_buffer().iter().collect();
+/// assert_eq!(to_vec, vec![false, true, false]);
+/// ```
+pub fn list_contains(array: &dyn Array, value: Scalar) -> VortexResult<Mask> {
```

Comment:
I think we should stick to returning Arrays here

Reply:
I think a common usecase for this function is that a user will want to call `filter` with the result, so a Mask made more sense to me. Converting between Array and Mask is a bit awkward right now

Reply:
`Mask::try_from(array)`?

Reply:
I still think the result should be an Array, just because I'm moving the compute functions towards only having array/scalar inputs/outputs. In this world, filter _may_ just become a function on ArrayImpl, rather than a compute function. Not entirely sure.